src/test/ui/unsized-locals/borrow-after-move.stderr

```diff
@@ -45,12 +45,12 @@ LL |         println!("{}", &y);
 error[E0382]: borrow of moved value: `x`
   --> $DIR/borrow-after-move.rs:39:24
    |
+LL |         let x = "hello".to_owned().into_boxed_str();
+   |             - move occurs because `x` has type `std::boxed::Box<str>`, which does not implement the `Copy` trait
```

Comment:
So... this is a bit suboptimal, in that the move is really occurring because `foo` was invoked on `*x` -- i.e., the type `str` -- but the new IR makes the borrow checker see it otherwise. I'm not sure how easy/hard that would be to fix. Probably not worth worrying about, I think.